src/WebJobs.Script.Analyzers/WebJobsAttributeAnalyzer.cs

```diff
@@ -0,0 +1,141 @@
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            context.EnableConcurrentExecution();
+
+            VerifyWebJobsLoaded();
+
+            // Analyze method signatures.
+            context.RegisterSyntaxNodeAction(AnalyzeMethodDeclarationNode, SyntaxKind.MethodDeclaration);
+
+            // Hook compilation to get the assemblies' references and build the WebJob tooling interfaces.
+            context.RegisterCompilationStartAction(AnalyzeCompilation);
+        }
+
+        private void AnalyzeCompilation(CompilationStartAnalysisContext context)
+        {
+            var compilation = context.Compilation;
+
+            AssemblyCache.Instance.Build(compilation);
+            _tooling = AssemblyCache.Instance.Tooling;
+
+            // cast to PortableExecutableReference which has a file path
+            var x1 = compilation.References.OfType<PortableExecutableReference>().ToArray();
+            var webJobsPath = (from reference in x1
+                               where IsWebJobsSdk(reference)
+                               select reference.FilePath).SingleOrDefault();
+
+            if (webJobsPath == null)
+            {
+                return; // Not a WebJobs project.
+            }
+        }
+
+        private bool IsWebJobsSdk(PortableExecutableReference reference)
+        {
+            if (reference.FilePath.EndsWith("Microsoft.Azure.WebJobs.dll"))
+            {
+                return true;
+            }
+            return false;
+        }
+
+        // This is called extremely frequently
+        // Analyze the method signature to validate binding attributes + types on the parameters
+        private void AnalyzeMethodDeclarationNode(SyntaxNodeAnalysisContext context)
+        {
+            if (_tooling == null) // Not yet initialized
+            {
+                return;
+            }
+
+            var methodDecl = (MethodDeclarationSyntax)context.Node;
+            var methodName = methodDecl.Identifier.ValueText;
+
+            if (!HasFunctionNameAttribute(context, methodDecl))
+            {
+                return;
+            }
+
+            // TODO: Attribute validation
+        }
+
+        // First argument to the FunctionName ctor.
+        private string GetFunctionNameFromAttribute(SemanticModel semantics, AttributeSyntax attributeSyntax)
+        {
+            foreach (var arg in attributeSyntax.ArgumentList.Arguments)
+            {
+                var val = semantics.GetConstantValue(arg.Expression);
+                if (!val.HasValue)
+                {
+                return null;
+                }
+                return val.Value as string;
+            }
+            return null;
+        }
+
+        // Does the method have a [FunctionName] attribute?
+        // This provides a quick check before we get into the more intensive analysis work.
```

Comment:
nit: Are these comments reflecting the latest version of the method?

Reply:
To the best of my knowledge, it did. I'm updating it to reflect the changes I'm pushing.